plugin.py

```diff
@@ -123,26 +127,27 @@ def on_pre_server_command(self, command: Mapping[str, Any], done_callback: Calla
             sublime.error_message(
                 'Cannot run executable "{}": You need to install the "Terminus" package and then restart Sublime Text'.format(output["kind"]))
             return False
-        if output["args"]["overrideCargo"]:
-            cargo_path = output["args"]["overrideCargo"]
-        else:
-            if not shutil.which("cargo"):
-                sublime.error_message('Cannot find "cargo" on path.')
-                return False
-            cargo_path = '"{}"'.format(shutil.which("cargo"))
-        command_to_run = [cargo_path] + output["args"]["cargoArgs"] + \
-            output["args"]["cargoExtraArgs"]
-        print(command_to_run)
-        cmd = " ".join(command_to_run)
-        args = {
-            "title": output["label"],
-            "shell_cmd": cmd,
-            "cwd": output["args"]["workspaceRoot"],
-            "auto_close": get_setting(view, "terminus_auto_close", False)
-        }
-        if get_setting(view, "terminus_use_panel", False):
-            args["panel_name"] = output["label"]
-        window.run_command("terminus_open", args)
+
+        for output in cargo_commands:
+            if output["args"]["overrideCargo"]:
+                cargo_path = output["args"]["overrideCargo"]
+            else:
+                if not shutil.which("cargo"):
```

Comment:
It would make sense to store this in a variable before the loop to avoid repeating the code and calling multiple times  (it's a fairly expensive IO operation)